mycli/main.py

```diff
@@ -811,16 +811,29 @@ def format_output(self, title, cur, headers, expanded=False,
             output.append(title)
 
         if cur:
-            rows = list(cur)
+            column_types = None
+            if hasattr(cur, 'description'):
+                def sanitize(col):
+                    if col is converters.through:
+                        return text_type
+                    else:
+                        return col
+                column_types = [sanitize(converters.decoders[col[1]])
+                                for col in cur.description]
+
+            if max_width is not None:
+                cur = list(cur)
```

Comment:
Auto vertical output has the potential to re-evaluate the cursor after the first try of horizontal output. This isn't possible without re-running the query which I find scary because a query might contain mutations.